### DISCUSSION TITLE
Support for the "OpenTelemetry" standard

### DISCUSSION THREAD

**Discussion by ziomill** [logfellow/logstash-logback-encoder#793] [Ideas]
Hello,

I'm considering the adoption of the OpenTelemetry standard and I'm planning to use the "magic" logstash-logback-encoder in order to emit logs.

Anyway, at the moment, the **logback** auto instrumentation provided by the OpenTelemetry libs doesn't support any encoder:
_**the javaagent instruments the logback Logger interface and captures the logs regardless of what appender or encoder you are using**_

I have implemented a little PoC and started a thread about this issue on the OpenTelemetry github:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/5496

Considering that OpenTelemetry is going to be the observability "de-facto" standard in the CNCF , it would be very useful if the **logstash-logback-encoder** could be supported for auto instrumentation.

**Comment by philsttr** [comment]
> it would be very useful if the logstash-logback-encoder could be supported for auto instrumentation.

Since auto instrumentation is done on the https://github.com/open-telemetry/opentelemetry-java-instrumentation side, I believe some changes will need to be done there.

Specifically, the OTel agent's logic for [mapping an ILoggingEvent to an OTel SDK LogBuilder](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/0302e9b921ba1e110b8cca66293f91bf1fb13253/instrumentation/logback/logback-appender-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/logback/appender/v1_0/LogbackHelper.java#L43-L84) would need to either :
* be aware of logstash-logback-encoder's structured arguments, and map them to the LogBuilder
  OR 
* provide some sort of plugin mechanism for others libraries to contribute to the LogBuilder (e.g. via [ServiceLoader](https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html) or some other dynamic discovery mechanism)

This could work with some sort of default set of JSON providers.  Or optionally OTel would need to provide some way to configure additional JSON providers

I don't think there's anything that can be done in the logstash-logback-encoder repository for this without changes on the opentelemetry-java-instrumentation side.

**Reply by ziomill** [reply]
@philsttr I agree with you.
What I was asking is if there is a plan to "contribute" to the OpenTelemetry in order to support the logstash encoder for logback.

As you said, the job should be done as a contribution on the OpenTelemetry side --> https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/CONTRIBUTING.md

Considering that the OpenTelemetry will soon become the observability standard in the CNCF world, I think that this would be a great improvement for the logback-logstash project!
